obspy/io/stationxml/core.py

```diff
@@ -1331,6 +1373,60 @@ def _format_time(value):
         return value.strftime("%Y-%m-%dT%H:%M:%S.%f+00:00")
 
 
+def _read_extra(element, obj):
+    """
+    Add information stored in custom tags/attributes in obj.extra.
+    """
+    # search all namespaces in current scope
+    for prefix, ns in element.nsmap.items():
+        # skip any fdsn namespaces,
+        # we're not interested in StationXML defined tags here
+        if ns.startswith("http://www.fdsn.org/") or \
+                ns.startswith("http://www.w3.org/"):
+            continue
+        # process all elements of this custom namespace, if any
+        for el in element.iterfind("{%s}*" % ns):
+            # remove namespace from tag name
+            _, name = el.tag.split("}")
+            value = el.text
+            try:
+                extra = obj.setdefault("extra", AttribDict())
+            # object is not based on AttribDict..
+            except AttributeError:
+                if hasattr(obj, "extra"):
+                    extra = obj.extra
+                else:
+                    extra = AttribDict()
+                    obj.extra = extra
+            etree.register_namespace(prefix, ns)
+            extra[name] = {'value': value,
+                           'namespace': '%s' % ns}
+            if el.attrib:
+                extra[name]['attrib'] = el.attrib
+    # process all attributes of custom namespaces, if any
+    for key, value in element.attrib.items():
+        # no custom namespace
+        if "}" not in key:
+            continue
+        # separate namespace from tag name
+        ns, name = key.lstrip("{").split("}")
```

Comment:
Better use this here as it will also work in some potentially unexpected cases with escaped braces and what not:

```python
_t = etree.QName(key)
ns, name = _t.namespace, _t.localname
```